src/java/us/kbase/mobu/runner/ModuleRunner.java

```diff
@@ -0,0 +1,289 @@
+            pw.println("job_service_url = " + jobSrvUrl);
+            pw.println("workspace_url = " + wsUrl);
+            pw.println("shock_url = " + shockUrl);
+            pw.println("kbase_endpoint = " + kbaseEndpoint);
+        } finally {
+            pw.close();
+        }
+        ////////////////////////////////// Preparing input.json ///////////////////////////////////
+        String jsonString;
+        if (inputFile != null) {
+            jsonString = FileUtils.readFileToString(inputFile);
+        } else if (inputJson != null) {
+            jsonString = inputJson;
+        } else if (stdin) {
+            ByteArrayOutputStream baos = new ByteArrayOutputStream();
+            IOUtils.copy(System.in, baos);
+            jsonString = new String(baos.toByteArray(), Charset.forName("utf-8"));
+        } else {
+            throw new IllegalStateException("No one input method is used");
+        }
+        jsonString = jsonString.trim();
+        if (!jsonString.startsWith("["))
+            jsonString = "[" + jsonString + "]";  // Wrapping one argument by array
+        if (verbose)
+            System.out.println("Input parameters: " + jsonString);
+        Map<String, Object> rpc = new LinkedHashMap<String, Object>();
+        rpc.put("version", "1.1");
+        rpc.put("method", methodName);
+        rpc.put("params", UObject.getMapper().readValue(jsonString, Object.class));
+        rpc.put("context", new LinkedHashMap<String, Object>());
+        UObject.getMapper().writeValue(new File(workDir, "input.json"), rpc);
+        ////////////////////////////////// Starting callback service //////////////////////////////
+        int callbackPort = NetUtils.findFreePort();
+        URL callbackUrl = CallbackServer.getCallbackUrl(callbackPort);
+        Server jettyServer = null;
+        if (callbackUrl != null) {
+            if( System.getProperty("os.name").startsWith("Windows") ) {
+                JsonServerSyslog.setStaticUseSyslog(false);
+                JsonServerSyslog.setStaticMlogFile("callback.log");
+            }
+            CallbackServerConfig cfg = new CallbackServerConfigBuilder(
+                    new URL(kbaseEndpoint), callbackUrl, runDir.toPath(),
+                    new LineLogger() {
+                        @Override
+                        public void logNextLine(String line, boolean isError) {
+                            //do nothing, SDK callback server doesn't use a logger
+                        }
+                    }).build();
+            ModuleRunVersion runver = new ModuleRunVersion(
+                    new URL("https://fakefakefakefakefake.com"),
+                    new ModuleMethod("use_set_provenance.to_set_provenance_for_tests"),
+                    "aaaaaaaaaaaaaaaaaaaaaaaaaaaaaaaa", "0.0.0", "dev");
+            JsonServerServlet catalogSrv = new SDKCallbackServer(
+                    auth, cfg, runver, new ArrayList<UObject>(),
+                    new ArrayList<String>());
```

Comment:
Same for the input parameters. Also, it'd be good to allow specifying a list of object references to set in the provenance in the CLI